pkg/supportbundle/supportbundle.go

```diff
@@ -403,3 +407,21 @@ func CreateSupportBundleAnalysis(appID string, archivePath string, bundle *types
 
 	return nil
 }
+
+func GetSpecSecretsMatchingLabel(client kubernetes.Interface, labelSelector string, namespace string, key string) (*corev1.Secret, error) {
```

Comment:
Is it possible use the function exported by troubleshoot rather than define it again in kots?

Reply:
In current version of kots, it is using `github.com/replicatedhq/troubleshoot v0.45.0` which only can return the spec string of secret. However, in our kots following steps, it needs the labels of secret.
```
existingSecret.Data[SpecDataKey] = renderedSpec
existingSecret.ObjectMeta.Labels = labels
```
For now, without updating the kots version to the latest, we need add this function. 
I agree with you that we should put this function back to troubleshoot and imported it. I have tested in my local, it is possible and I will create a pr to add this function in latest troubleshoot.